Piwik.Tracker.Samples/PiwikTrackerSamples.cs

```diff
@@ -84,46 +83,46 @@ static private void RecordCustomVariables()
             piwikTracker.SetCustomVariable(1, "pagevar1", "pagevalue1", Scopes.Page);
             piwikTracker.SetCustomVariable(2, "pagevar2", "pagevalue2", Scopes.Page);
 
-            var response = piwikTracker.DoTrackPageView("Document title of current page view");
+            var response = await piwikTracker.TrackPageViewAsync("Document title of current page view");
 
             DisplayDebugInfo(response);
         }
 
         /// <summary>
         /// Records 2 custom dimensions
         /// </summary>
-        static private void RecordCustomDimensions()
+        private static async Task RecordCustomDimensionsAsync()
         {
-            var piwikTracker = new PiwikTracker(SiteId, PiwikBaseUrl);
+            var piwikTracker = new PiwikTracker(HttpClient, SiteId, PiwikBaseUrl);
             piwikTracker.SetUserAgent(UA);
 
             piwikTracker.SetCustomTrackingParameter("dimension1", "value1");
             piwikTracker.SetCustomTrackingParameter("dimension2", "value2");
 
-            var response = piwikTracker.DoTrackPageView("Document title of current page view");
+            var response = await piwikTracker.TrackPageViewAsync("Document title of current page view");
 
             DisplayDebugInfo(response);
         }
 
         /// <summary>
         ///  Records a simple page view with a specified document title
         /// </summary>
-        static private void RecordSimplePageView()
+        private static async Task RecordSimplePageViewAsync()

```

Comment:
I've uncommented the methods and made them work using Task.Run().GetAwaiter().GetResult(). Usually doing so is a bad idea due to possible deadlock. However, it is ok for console app, as the Main method doesn't run in the thread pool, thus doesn't need to compete for available threads (if I'm not wrong).